installer/installer.py

```diff
@@ -1197,6 +1210,123 @@ def configure_node_settings(self) -> bool:
             logging.exception(f"Failed to configure cheqd-noded settings. Reason: {e}")
             return False
 
+    def _select_working_rpc_endpoint(self, chain: str) -> str:
+        try:
+            endpoints = []
+            if chain == "testnet":
+                endpoints = [TESTNET_RPC_ENDPOINT_EU, TESTNET_RPC_ENDPOINT_AP]
+            else:
+                endpoints = [MAINNET_RPC_ENDPOINT_EU, MAINNET_RPC_ENDPOINT_AP]
+
+            for endpoint in endpoints:
+                try:
+                    req = request.Request(f"{endpoint}/status")
+                    with request.urlopen(req, timeout=10) as resp:
+                        if resp.getcode() == 200:
+                            return endpoint
+                except Exception:
+                    continue
+        except Exception as e:
+            logging.exception(f"Could not select a working RPC endpoint. Reason: {e}")
+        return ""
```

Comment:
This approach tries endpoints in the order they are listed, which may or may not be ideal.

Consider a latency / physical proximity approach, favouring smallest ping:


```suggestion
    def _select_fastest_rpc_endpoint(self, chain: str) -> str:
    endpoints = []
    if chain == "testnet":
        endpoints = [TESTNET_RPC_ENDPOINT_EU, TESTNET_RPC_ENDPOINT_AP]
    else:
        endpoints = [MAINNET_RPC_ENDPOINT_EU, MAINNET_RPC_ENDPOINT_AP]

    best_endpoint = ""
    best_latency = float("inf")

    for endpoint in endpoints:
        try:
            req = request.Request(f"{endpoint}/status")
            start_time = time.time()
            with request.urlopen(req, timeout=10) as resp:
                if resp.getcode() == 200:
                    latency = time.time() - start_time
                    if latency < best_latency:
                        best_latency = latency
                        best_endpoint = endpoint
        except Exception:
            continue

    return best_endpoint
```

Reply:
That makes sense, but I don't know if that's the ideal approach.
State Sync requires 2 RPC servers configured, so I want to add both endpoints to distribute traffic evenly.

I modified this function to check endpoints' health first, and if there's only one healthy one, it would just use the same endpoint twice (there's no additional check in CometBFT).

Alternatively, we can just use the top-level `https://rpc.cheqd.net:443` endpoint twice and leave the geo-location/proximity calculations to Cloudflare.

I'm fine with both options, just having 2 different RPC endpoints seemed like a cleaner configuration, but it doesn't make a huge difference.